File: extensions/nyno-html-to-markdown/command.py
# extensions/html_to_markdown/command.py
from html_to_markdown import convert

# Nyno extension: Convert HTML to Markdown using the Rust-based html-to-markdown library

def nyno_html_to_markdown(args, context):
    # Determine input HTML
        # args[0] may be a string or a list of HTML strings
    first_arg = args[0] if args else ""
    if isinstance(first_arg, list):
        html_inputs = first_arg
    else:
        html_inputs = [first_arg]

    # Determine output context key
    if "set_context" in context:
        set_name = context["set_context"]
    else:
        set_name = "prev"

    try:
        # Convert HTML → Markdown
                        # Convert each HTML item to Markdown
        md_list = []
        for item in html_inputs:
            md_list.append(convert(item))

        # Preserve original type: single string in → single string out
        if isinstance(first_arg, list):
            md = md_list
        else:
            md = []
        for item in html_inputs:
            md.append(convert(item))

        # Store result
        context[set_name] = md

        return 0

    except Exception as e:
        # Store error
        context[f"{set_name}.error"] = {"error": str(e)}
        return 1

File: extensions/nyno-markdown-to-html/command.js
// extensions/markdown-to-html/command.js
import MarkdownIt from 'markdown-it';

export function nyno_markdown_to_html(args, context) {
    const input = args[0];
    if (!input) {
        const errMsg = "No Markdown content provided in args[0]";
        console.error(errMsg);
        const setName = "set_context" in context ? context["set_context"] : "prev";
        context[setName + ".error"] = { errorMessage: errMsg };
        return 1;
    }

    const md = new MarkdownIt();
    const setName = "set_context" in context ? context["set_context"] : "prev";

    try {
        if (Array.isArray(input)) {
            // Handle list of Markdown strings
            context[setName] = input.map(item => {
                const html = md.render(item);
                return html;
            });
        } else {
            // Handle single Markdown string
            const html = md.render(input);
            context[setName] = html;
        }

        return 0; // success
    } catch (err) {
        console.error(err);
        context[setName + ".error"] = { errorMessage: err.message };
        return 1; // failure
    }
}

